app/kuma-dp/pkg/dataplane/metrics/server.go

```diff
@@ -58,16 +60,36 @@ type (
 	OtelMutator    func(in io.Reader) ([]*io_prometheus_client.MetricFamily, error)
 )
 
-type QueryParametersModifier func(queryParameters url.Values) string
+type QueryParametersModifier func(queryParameters url.Values) url.Values
 
-func RemoveQueryParameters(_ url.Values) string {
-	return ""
+func RemoveQueryParameters(_ url.Values) url.Values {
+	return url.Values{}
 }
 
-func AddPrometheusFormat(queryParameters url.Values) string {
+func AddPrometheusFormat(queryParameters url.Values) url.Values {
 	queryParameters.Add("format", "prometheus")
 	queryParameters.Add("text_readouts", "")
-	return queryParameters.Encode()
+	return queryParameters
+}
+
+func AddUsedOnlyParameter(sidecar *v1alpha12.Sidecar) func(queryParameters url.Values) url.Values {
```

Comment:
naming of this method does not suggest it will also add filters, maybe we can change it to something like `AddSidecarParameters`